CONTRIBUTING.md

```diff
@@ -0,0 +1,171 @@
+# Contributing guide
+
+We're always looking for new contributors and are happy to help you get started! Some ideas:
+
+- Add a test to a function, testing something that we haven't tested yet.
+- Improve a function's documentation.
+- Fix a bug or an edge case.
+- Work on [![issues labeled good first issue](https://img.shields.io/github/issues/remeda/remeda/good%20first%20issue?style=flat-square)](https://github.com/remeda/remeda/issues?q=is%3Aopen+is%3Aissue+label%3A%good+first+issue%22).
+- Work on [![issues labeled help wanted](https://img.shields.io/github/issues/remeda/remeda/help%20wanted?style=flat-square)](https://github.com/remeda/remeda/issues?q=is%3Aopen+is%3Aissue+label%3A%help+wanted%22).
+
+## Getting started
+
+We use `npm` as the package manager. We recommend using [`nvm`](https://github.com/nvm-sh/nvm) to manage `Node.js` and `npm` versions.
+
+Install dependencies:
+
+```bash
+git clone git@github.com:remeda/remeda.git
+cd remeda
+npm install
+```
+
+Run tests and watch for changes:
+
+```bash
+npm test # every test
+npm test -i src/file.test.ts # specific test
+```
+
+We have a pre-commit script that will format and lint the project, but if you want to run it manually:
+
+```bash
+npm run prettier # format
+npm run lint # lint
+```
+
+## Documentation
+
+Most of the content in the documentation is generated from the JSDoc comments. If you want to work on the design of the docs website or the migration guides, see [`docs/README.md`](https://github.com/remeda/remeda/blob/main/docs/README.md).
+
+## Guidelines
+
+### Adding a new function
+
+When adding a new function, remember to:
+
+- Add a JSDoc comment with a description, parameters, signature, an example, and tags.
+- Add tests for runtime, typing, data-first, and data-last forms.
+  - For data-last tests, prefer tests using `pipe`, as this better matches real-world usage.
+- Add exports to `src/index.ts`.
+- Check for equivalent functions in [Lodash](https://lodash.com/docs/4.17.15) and [Ramda](https://ramdajs.com/docs/), and add them to `docs/src/content/mapping`.
+
+### Writing types
+
+TODO: using `type-fest`, prefer `<T extends IterableContainer>(data: T)` over `<T>(data: ReadonlyArray<T>)`, and `<T extends object>(data: T)` over `<T>(data: Readonly<Record<string, T>>)`
```

Comment:
I don't understand this TODO, and the sentence that follows, it's unrelated to type-fest...

Reply:
i filled in the TODO, does it make more sense now?